lib/prometheus_exporter/client.rb

```diff
@@ -4,16 +4,16 @@
 require 'thread'
 
 module PrometheusExporter
-
   class Client
     class RemoteMetric
       attr_reader :name, :type, :help
 
-      def initialize(name:, help:, type:, client:)
+      def initialize(name:, help:, type:, client:, opts: {})
```

Comment:
`opts: nil`

Then `opts || {}` is not needed so we allocate one less hash.

Reply:
if the caller explicitly passes in nil to initialize, this would still be necessary to ensure opts is a hash.

Reply:
not until way later, this check can be done in the collector, the remote metric can simply not send that field across if it is nil. (at least I think) :thinking:

Reply:
yeah, defaulting to nil now and setting only in the collector. we also added some tests for it. i think the test failure in travis was intermittent. this pull request is ready from our perspective. we can squash the commits if you want.